### PR TITLE
SAIC-180 VM evacuation

### PR DESCRIPTION
**Pull request by roman-verchikov**
- script reads series of actions from db and performs
    novaclient.servers.live_migrate call for each action

### CODE REVIEW

**Comment by Wasapl** [comment]
Merged build finished. Test PASSED.

File: configs/config.ini [hunk 1]
```diff
@@ -69,6 +69,8 @@ backend = iscsi
 convert_diff_file = qcow2
 convert_ephemeral_disk = qcow2
 host_eph_drv = <src_host_epehem_drv>
+block_migration =
```
Comment (miarmak) [inline configs/config.ini]: Maybe we'll put here default values in config too?

File: evacuation/actions/evacuate_vms.py [hunk 2]
```diff
@@ -0,0 +1,37 @@
+import json
```
Comment (miarmak) [inline evacuation/actions/evacuate_vms.py]: Copyright?

File: evacuation/evacuation_chain.py [hunk 3]
```diff
@@ -0,0 +1,12 @@
+from evacuation.actions import evacuate_vms
```
Comment (miarmak) [inline evacuation/evacuation_chain.py]: One more copyright?

File: fabfile.py [hunk 4]
```diff
@@ -24,6 +24,8 @@
 from dry_run import chain
 from condensation import process
 from condensation.scripts import nova_collector as nova_collector_module
+from evacuation import evacuation_chain
+>>>>>>> Implements SAIC-180 : evacuation script
```
Comment (miarmak) [inline fabfile.py]: I guess this line is redundant.

**Comment by Wasapl** [comment]
Merged build finished. Test PASSED.

File: cloudferrylib/os/compute/instances.py [hunk 5]
```diff
@@ -19,3 +23,34 @@ def update_user_ids_for_instance(db, instance_id, user_id):
...
+    ))
+    nova_client.servers.live_migrate(
+        server=vm_id,
+        host=hostname,
+        block_migration=config.compute.block_migration,
+        disk_over_commit=config.compute.disk_overcommit
+    )
+
+
+def cobalt_live_migrate_vm(nova_client, config, vm_id, hostname):
```
Comment (miarmak) [inline cloudferrylib/os/compute/instances.py]: Where is the `config` variable used?

File: cloudferrylib/os/compute/instances.py [hunk 6]
```diff
@@ -19,3 +23,34 @@ def update_user_ids_for_instance(db, instance_id, user_id):
...
+    LOG.debug("migrating {vm} to {host}".format(
+        vm=vm_id,
+        host=hostname
+    ))
+    nova_client.servers.cobalt_migrate(
```
Comment (miarmak) [inline cloudferrylib/os/compute/instances.py]: Are you sure novaclient provides such method?

File: cloudferrylib/os/compute/instances.py [hunk 7]
```diff
@@ -19,3 +23,34 @@ def update_user_ids_for_instance(db, instance_id, user_id):
...
+        server=vm_id,
+        dest=hostname
+    )
+
+
+def incloud_live_migrate(nova_client, config, vm_id, destination_host):
+    if config.migrate.incloud_live_migration == 'nova':
```
Comment (miarmak) [inline cloudferrylib/os/compute/instances.py]: There is no such option in our config. Add it there please.

**Comment by Wasapl** [comment]
Merged build finished. Test PASSED.

**Comment by Wasapl** [comment]
Merged build finished. Test PASSED.

**Comment by Wasapl** [comment]
Merged build finished. Test PASSED.

**Comment by Wasapl** [comment]
Can one of the admins verify this patch?

**Comment by Wasapl** [comment]
Build finished.
Unittest success

**Comment by Wasapl** [comment]
Unittest success

**Comment by Wasapl** [comment]
Build finished.
Unittest success

**Comment by Wasapl** [comment]
Unittest success

**Comment by Wasapl** [comment]
Build finished.
Unittest success

**Comment by Wasapl** [comment]
Unittest success

**Comment by Wasapl** [comment]
Build finished.
Unittest success

**Comment by Wasapl** [comment]
Unittest success

**Comment by roman-verchikov** [comment]
Verified on production env with cobalt - works fine

**Comment by Wasapl** [comment]
Build finished.
Unittest success

**Comment by Wasapl** [comment]
Unittest success

**Comment by Wasapl** [comment]
Build finished.
Unittest success

**Comment by Wasapl** [comment]
Unittest success

**Comment by Wasapl** [comment]
Build finished.
Unittest success

**Comment by Wasapl** [comment]
Unittest success

**Pull request merged**